### PR TITLE
completed task 5-get-articles

### PR DESCRIPTION
**Pull request by zaw-tun**
Completed task 5 GET /api/articles

### CODE REVIEW

**Review by nc-ali-abdallah — COMMENTED** [review]
Awesome job on this ticket.
It may of been mentioned previously but I would really advise taking a look into toMatchObjects and expect.objectContaining for future tests.

File: __tests__/api.test.js [hunk 1]
```diff
@@ -90,3 +93,27 @@ describe("GET /api/articles/:article_id", () => {
...
+        body.articles.forEach((article) => {
+          expect(article).toHaveProperty("author");
+          expect(article).toHaveProperty("title");
+          expect(article).toHaveProperty("article_id");
+          expect(article).toHaveProperty("topic");
+          expect(article).toHaveProperty("created_at");
+          expect(article).toHaveProperty("votes");
+          expect(article).toHaveProperty("article_img_url");
+          expect(article).toHaveProperty("comment_count");
+          expect(article.body).toBe(undefined);
```
Comment (nc-christian-dutton) [inline __tests__/api.test.js:115]: I think checking expect(article.body).not.toHaveProperty("body") would be better than checking it's value is undefined!

File: models/articles.model.js [hunk 2]
```diff
@@ -10,3 +10,17 @@ exports.selectArticle = (article_id) => {
...
 };
+
+exports.selectArticles = () => {
+  return db
+    .query(
+      `
+        SELECT a.author, a.title, a.article_id, a.topic, a.created_at, a.votes, a.article_img_url, COUNT(c.comment_id) AS comment_count FROM articles AS a
+        LEFT JOIN comments AS c USING (article_id)
+        GROUP BY a.article_id ORDER BY a.created_at DESC;
+        `
```
Comment (nc-christian-dutton) [inline models/articles.model.js:21]: Great job with the COUNT and nice clean sql syntax 👍

File: __tests__/api.test.js [hunk 3]
```diff
@@ -90,3 +93,27 @@ describe("GET /api/articles/:article_id", () => {
...
+
+describe("GET /api/articles", () => {
+  it("status: 200, responds with all articles sorted by date in descending order", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then((response) => {
+        const { body } = response;
+        expect(body.articles.length).toBe(13);
+        expect(body.articles).toBeSortedBy("created_at", { descending: true });
```
Comment (nc-christian-dutton) [inline __tests__/api.test.js:105]: Good job getting jest-sorted in here!

**Pull request merged**